android/src/main/java/com/oney/WebRTCModule/UVCVideoCapturer.java

```diff
@@ -0,0 +1,117 @@
+
+public class UVCVideoCapturer implements CameraVideoCapturer {
+
+    private SurfaceTextureHelper surfaceTextureHelper;
+    private CapturerObserver capturerObserver;
+    private CameraUvcStrategy cameraUvcStrategy;
+    private PreviewDataCallback previewDataCallback;
+    private boolean isCapturerStarted;
+
+    @Override
+    public synchronized void initialize(SurfaceTextureHelper surfaceTextureHelper, Context context, CapturerObserver capturerObserver) {
+        this.surfaceTextureHelper = surfaceTextureHelper;
+        this.capturerObserver = capturerObserver;
+        this.previewDataCallback = new PreviewDataCallback(this.capturerObserver::onFrameCaptured);
+        this.cameraUvcStrategy = new CameraUvcStrategy(context);
+        this.cameraUvcStrategy.register();
+    }
+
+    @Override
+    public synchronized void startCapture(int width, int height, int fps) {
+        stopCapture();
+
+        Log.i(WebRTCModule.TAG, String.format("Starting uvc capture with %d x %d @ %d", width, height, fps));
+        this.capturerObserver.onCapturerStarted(true);
+        this.isCapturerStarted = true;
+        this.cameraUvcStrategy.addPreviewDataCallBack(this.previewDataCallback);
+        this.cameraUvcStrategy.startPreview(createCameraRequest(width, height), this.surfaceTextureHelper.getSurfaceTexture());
+    }
+
+    @Override
+    public synchronized void stopCapture() {
+        Log.d(WebRTCModule.TAG, "Stopping uvc capture");
+        this.cameraUvcStrategy.removePreviewDataCallBack(this.previewDataCallback);
+        this.cameraUvcStrategy.stopPreview();
+        if (this.isCapturerStarted) {
+            this.capturerObserver.onCapturerStopped();
+            this.isCapturerStarted = false;
+        }
+    }
+
+    @Override
+    public synchronized void changeCaptureFormat(int width, int height, int fps) {
+        Log.d(WebRTCModule.TAG, String.format("Capture format changed to %d x %d @ %d", width, height, fps));
+        this.stopCapture();
+        this.startCapture(width, height, fps);
+    }
+
+    @Override
+    public synchronized void dispose() {
+        Log.d(WebRTCModule.TAG, "Disposing uvc capturer");
+        stopCapture();
+        this.cameraUvcStrategy.unRegister();
+    }
+
+    @Override
+    public boolean isScreencast() {
+        return false;
+    }
+
+    @Override
+    public void switchCamera(CameraSwitchHandler cameraSwitchHandler) {
+        cameraSwitchHandler.onCameraSwitchError("Not supported yet with uvc");
+    }
+
+    @Override
+    public void switchCamera(CameraSwitchHandler cameraSwitchHandler, String s) {
+        cameraSwitchHandler.onCameraSwitchError("Not supported yet with uvc");
+    }
+
+    private CameraRequest createCameraRequest(int width, int height) {
+        return new CameraRequest
+                .Builder()
+                .setPreviewWidth(width)
+                .setPreviewHeight(height)
+                .setAudioSource(CameraRequest.AudioSource.SOURCE_AUTO)

```

Comment:
Audio?

Reply:
Removed this to prevent confusion, since audio is not yet captured from the uvc device.